source/migration/Migration/CecilMigrator.cs

```diff
@@ -582,6 +571,41 @@ private bool CopyFile(string src, string dst)
 			return true;
 		}
 
+		private AssemblyDefinition ReadAssembly(AssemblyResolver resolver, string path)
+		{
+			var readerParams = new ReaderParameters
+			{
+				ReadSymbols = true,
+				ReadWrite = true,
+				AssemblyResolver = resolver,
+			};
+
+			var pdbPath = Path.ChangeExtension(path, "pdb");
+			var mdbPath = path + ".mdb";
+
+			if (File.Exists(pdbPath) || File.Exists(mdbPath))
+			{
+				// if there are symbols, then try using them
+				try
+				{
+					return AssemblyDefinition.ReadAssembly(path, readerParams);
+				}
+				catch (Exception ex)
+				{
+					// do nothing
+					LogMessage($"  Debug symbols were found for the assembly, but they couldn't be used: " + ex.Message);
+				}
+			}
+			else
+			{
+				LogVerboseMessage($"  No debug symbols found for the assembly.");
+			}
+
+			// just ignore all symbols, there is nothing that can be done about bad symbols
+			readerParams.ReadSymbols = false;
+			return AssemblyDefinition.ReadAssembly(path, readerParams);
+		}
```

Comment:
This looks the same we've been using: https://github.com/xamarin/java.interop/blob/377c4c7ab50d1f25082c7079214b8075eb25de35/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs#L144-L172

We use this `DirectoryAssemblyResolver` any time we use Mono.Cecil during Xamarin.Android's build.